test/e2e/fixtures/imported-account/state.json

```diff
@@ -489,6 +492,6 @@
     }
   },
   "meta": {
-    "version": 40
+    "version": 63
```

Comment:
This change in version broke the e2e tests. The `participateInMetaMetrics` state moved in migration 49, so now that modal is showing at the start of each test using this.

For now you should be able to leave this as it was before I think? But in general we really ought to setup some script to run migrations on fixture files to make updating their versions easier.

Reply:
So I'm trying to figure out the best way to resolve an issue caused by `primaryTransaction` on the transactionGroup being passed to useIncrementedGasFees being `undefined` sometimes in a couple of e2e tests... I'm not entirely sure why. I originally solved this by add a default to the primaryTransaction being destructured in useIncrementedGasFees:  `const { primaryTransaction = {} } = transactionGroup;` but I should probably dig in a bit more into the cause.

Reply:
That's curious. This state log has _no_ transactions, so there should be no transaction items shown. Which test did you see this failure in?

Reply:
Sorry yes I wasn't clear, so I dug in a bit deeper and primaryTransaction (and initial transaction for that matter) aren't on the transaction object (which becomes the transactionGroup object) until in the transaction-list, so there won't be a primaryTransaction when, for instance the user clicks on the edit gas modal in the send flow, as the code stands. So in [my latest commit](https://github.com/MetaMask/metamask-extension/pull/12069/commits/50481bd0139958710c0bfd7d43ac42168c33df69) I check if there is a primaryTransaction defined on the transactionsGroup object and, if not, put the root level txParams in as the primaryTransaction. This may not be exactly the way we want to do it, but its the right direction.

Reply:
@Gudahtt

Reply:
That doesn't make much sense to me :thinking: If it's a transaction group, it has to have a `primaryTransaction`. If it's not, then the parameters are mismatched here.